src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -179,11 +179,17 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
     // Related Bug: https://bugs.launchpad.net/mixxx/+bug/1983204
     // In order to not hit the rate limits and respect their rate limiting rule.
     // We are going to delay every request by one second.
-
-    while (m_timer.elapsed(true) < kDelayDuration) {
+    if (m_timer.elapsed(true).toIntegerSeconds() >= 1) {
```

Comment:
Please get familiar how timers in conjunction with the Qt event loop are supposed to be used, i.e. connect their signals to slots. No sleeps, no hot loops. You need to do all the state management in your class manually. Getting familiar on how finite state machines (FSM) work helps.

https://doc.qt.io/qt-5/qtimer.html

Reply:
I will take a look, do my search and try to find the best way possible. As you mentioned, no sleep - no hot loop. 

Thank you for this cool tip and the doc link. That will be really helpful for me.

Reply:
In this case we have the issue that a possible abort() signal is ignored during the sleep. 
This was actually already Ignoted her before due to the direct slotStart() call.

A QTimer::timeout() signal will solve both. It is put into the Qt message queue after a possible abort() signal.
So in case of abort the timeout() handled after the abort(). You need to stop the timer in this case that the process does not start over after abort(), and also expect timeouts() that may happen concurrently.